fvm/evm/handler/handler.go

```diff
@@ -134,45 +134,75 @@ func (h *ContractHandler) LastExecutedBlock() *types.Block {
 }
 
 // RunOrPanic runs an rlp-encoded evm transaction and
-// collects the gas fees and pay it to the coinbase address provided.
-func (h *ContractHandler) RunOrPanic(rlpEncodedTx []byte, coinbase types.Address) {
+func (h *ContractHandler) RunOrPanic(rlpEncodedTx []byte, gasFeeCollector types.Address) {
 	// capture open tracing span
 	defer h.backend.StartChildSpan(trace.FVMEVMRun).End()
 
-	res, err := h.run(rlpEncodedTx, coinbase)
+	// capture coinbase init balance
+	cb := h.AccountByAddress(types.CoinbaseAddress, true)
+	initCoinbaseBalance := cb.Balance()
+
+	res, err := h.run(rlpEncodedTx)
```

Comment:
what if the transaction itself sends funds to this address, for whatever reason, then I guess those funds end in the coinbase, which I guess is ok, just flagging for you to think about it

Reply:
Yeah thats the behviour , I was more worried about corner cases of gas fee calculation , so I took this approach to work on diff